updater/lib/dependabot/dependency_snapshot.rb

```diff
@@ -11,58 +11,34 @@
 # representing the output.
 module Dependabot
   class DependencySnapshot
-    # TODO: Enforce non-nil values for job_definition["base64_dependency_files"]
-    #       and job_definition["base_commit_sha"]
-    #
-    # We historically tolerate nil values for both these keys from the `job_definition`
-    # but it doesn't seem like we should. Rather than introduce a behaviour change
-    # as part of the change introducing this class, let's do it as a follow-up.
     def self.create_from_job_definition(job:, job_definition:)
-      decoded_dependency_files = job_definition.fetch("base64_dependency_files", []).map do |a|
+      decoded_dependency_files = job_definition.fetch("base64_dependency_files").map do |a|
         file = Dependabot::DependencyFile.new(**a.transform_keys(&:to_sym))
         file.content = Base64.decode64(file.content).force_encoding("utf-8") unless file.binary? && !file.deleted?
         file
       end
 
       new(
         job: job,
-        base_commit_sha: job_definition.fetch("base_commit_sha", nil),
+        base_commit_sha: job_definition.fetch("base_commit_sha"),
         dependency_files: decoded_dependency_files
       )
     end
 
-    attr_reader :base_commit_sha, :dependency_files
+    attr_reader :base_commit_sha, :dependency_files, :dependencies
+
+    private
 
     def initialize(job:, base_commit_sha:, dependency_files:)
       @job = job
       @base_commit_sha = base_commit_sha
       @dependency_files = dependency_files
-    end
-
-    def dependencies
-      return @dependencies if defined?(@dependencies)
 
       parse_files!
```

Comment:
💅 I think it makes sense to explicitly set `@dependencies` here.

Reply:
The diff doesn't really allow for this, but FWIW, I was referring to setting `@dependencies` in the initializer (and removing the setting of it from `#parse_files!`